feat(ia): advertising > display ads

jaredrethman opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

* New Advertising admin page.
* New sub menu pages; Display Ads (Full) & Sponsors (Initial).
* Migrate legacy advertising wizard to new IA location. 
* Migrated Add-Ons tab to Settings tab.
* Migrated Suppressions tab to Settings tab.
* Added titles to each tab and section.
* CSS changes to accommodate title additions and tighten up component spacing.

### How to test the changes in this Pull Request:

1. Checkout this code and build assets.
2. Navigate to _/wp-admin/admin.php?page=advertising-display-ads#/_
3. Observe each tab; Providers, Placement and Settings all load as expected.
4. On Settings tab confirm Suppressions and Plugins (previously Add-Ons) are present.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

jaredrethman committed:
feat(ia): display ads initial

jaredrethman committed:
Merge remote-tracking branch 'origin/epic/ia' into feat/ia-ads

jaredrethman committed:
Merge remote-tracking branch 'origin/epic/ia' into feat/ia-ads

jaredrethman committed:
refactor(ia): headings

jaredrethman committed:
refactor(ia): heading and newspack-card margins

jaredrethman committed:
revert(ia): prettier formatting

jaredrethman committed:
refactor(ia): remove `admin_notices` on wizard screen

jaredrethman committed:
Merge remote-tracking branch 'origin/epic/ia' into feat/ia-ads

dkoo commented on includes/wizards/advertising/class-advertising-sponsors.php:
```diff
@@ -0,0 +1,63 @@
+	 * @return string The wizard name.
+	 */
+	public function get_name() {
+		return esc_html__( 'Sponsors', 'newspack' );
```
```suggestion
		return \esc_html__( 'Sponsors', 'newspack-plugin' );
```

dkoo commented on includes/wizards/advertising/class-advertising-display-ads.php:
```diff
@@ -0,0 +1,561 @@
+	 * @return string The wizard name.
+	 */
+	public function get_name() {
+		return \esc_html__( 'Advertising', 'newspack' );
```
```suggestion
		return \esc_html__( 'Advertising', 'newspack-plugin' );
```

dkoo commented on includes/wizards/advertising/class-advertising-display-ads.php:
```diff
@@ -0,0 +1,561 @@
+			$ad_units = $configuration_manager->get_ad_units();
+		} catch ( \Exception $error ) {
+			$message = $error->getMessage();
+			$error   = new WP_Error( 'newspack_ad_units', $message ? $message : __( 'Ad Units failed to fetch.', 'newspack' ) );
```
```suggestion
			$error   = new WP_Error( 'newspack_ad_units', $message ? $message : __( 'Ad Units failed to fetch.', 'newspack-plugin' ) );

dkoo reviewed (changes requested):
Working as advertised (heh). 

The only thing I caught was the lack of whitespace between the ad units when configuring GAM:

**`trunk`**

<img width="1050" alt="Screenshot 2024-07-30 at 12 07 56 PM" src="https://github.com/user-attachments/assets/504da172-76a8-4100-ab09-703786367430">

**`feat/ia-ads`**

<img width="1044" alt="Screenshot 2024-07-30 at 12 08 11 PM" src="https://github.com/user-attachments/assets/7d900003-7c65-4369-ad81-3aad1311a752">

All other functionality works as expected, except the Plugin cards (e.g. the Configure links after installing—I expect this will be fixed in a future PR since the testing instructions here only mention that they should be appearing in the right place, right).

Something non-blocking I just noticed in this PR but haven't been looking out for: we have a mismatch in some of our translation text domains throughout the Newspack Plugin due to legacy code. The text domain should be `newspack-plugin`, matching the plugin slug, but there are many instances of `newspack` throughout. It's not a blocker at all, but when editing surrounding code I try to update those if I see them. I highlighted all the instances I saw in this changeset, but like I said I don't consider these blockers.

jaredrethman committed:
Update includes/wizards/advertising/class-advertising-display-ads.php

Co-authored-by: Derrick Koo <dkoo@users.noreply.github.com>

jaredrethman committed:
Update includes/wizards/advertising/class-advertising-sponsors.php

Co-authored-by: Derrick Koo <dkoo@users.noreply.github.com>

jaredrethman committed:
Update includes/wizards/advertising/class-advertising-display-ads.php

Co-authored-by: Derrick Koo <dkoo@users.noreply.github.com>

jaredrethman committed:
feat(ia): added in page anchor navigation

jaredrethman committed:
Merge branch 'feat/ia-ads' of github.com:Automattic/newspack-plugin into feat/ia-ads

jaredrethman commented:
@dkoo thanks for the feedback, responses below.

> Working as advertised (heh).

🤣 

> The only thing I caught was the lack of whitespace between the ad units when configuring GAM:

Added back.

> All other functionality works as expected, except the Plugin cards (e.g. the Configure links after installing—I expect this will be fixed in a future PR since the testing instructions here only mention that they should be appearing in the right place, right).

Have added in page anchor navigation.

> Something non-blocking I just noticed in this PR but haven't been looking out for: we have a mismatch in some of our translation text domains throughout the Newspack Plugin due to legacy code. The text domain should be newspack-plugin, matching the plugin slug, but there are many instances of newspack throughout. It's not a blocker at all, but when editing surrounding code I try to update those if I see them. I highlighted all the instances I saw in this changeset, but like I said I don't consider these blockers.

Yeah, I've noticed these too. I've replaced all text domains in the newer stuff, but neglected to do so for the migration of the older stuff. Will keep an eye out for this going forward.

dkoo commented:
> Have added in page anchor navigation.

Works well after a page refresh, but since some of the settings sections aren't rendered if we don't have the plugin requirements, if you click "Configure" immediately after installing it will throw a JS error about the element being undefined. Maybe we could trigger a page refresh after the install to get around that?

jaredrethman committed:
refactor: pr feedback - reload browser after install/uninstall

jaredrethman commented:
> Works well after a page refresh, but since some of the settings sections aren't rendered if we don't have the plugin requirements, if you click "Configure" immediately after installing it will throw a JS error about the element being undefined. Maybe we could trigger a page refresh after the install to get around that?

Nice catch. Have added `shouldRefreshAfterUpdate` property to both plugin objects.

dkoo reviewed (approved):
Looking good now, thanks for the updates!
